lib/tools.js

```diff
@@ -998,20 +998,20 @@ function getRepositoryFile(urlOrPath, additionalInfo, callback) {
             } else {
                 getJson(urlOrPath, agent, sources => {
                     if (sources) {
-                        for (const s in sources) {
-                            if (sources.hasOwnProperty(s) && additionalInfo[s] && additionalInfo[s].published) {
+                        for (const s of Object.keys(sources)) {
+                            if (additionalInfo[s] && additionalInfo[s].published) {
                                 sources[s].published = additionalInfo[s].published;
                             }
                         }
                         setImmediate(() =>
                             _getRepositoryFile(sources, path, err => {
-                                err && console.error('[' + new Date() + '] ' + err);
+                                err && console.error(`[${new Date()}] ${err}`);
                                 typeof callback === 'function' && callback(err, sources, actualSourcesHash);
                             }));
                     } else {
-                        if (typeof callback === 'function') {
-                            callback('Cannot read "' + urlOrPath + '"', {});
-                        }
+                        // return cached sources, because no sources found
```

Comment:
relevant part, maybe we should also return no error like the other methods do, which return cached sources

Reply:
The problem is that logging to console is invisible mainly ... so the error is then logged in the calling methid I think

Reply:
Or other ... we would need to check how the "calling code" works ... if they nly see "err" and then cancel execution we should not return error

Reply:
Ok should work with returning an error. all "external" users of getRepositoryFile method I found logged error when it was set, but then check for sources existence